src/activities/quizzes/attempts/QuizAttemptsEntity.js

```diff
@@ -106,6 +127,32 @@ export class QuizAttemptsEntity extends Entity {
 		return this._entity && this._entity.getSubEntityByClass(Classes.quizzes.attempts.retakeIncorrectOnly);
 	}
 
+	/**
+	 * @returns {object} quiz attempt conditions sub-entity
+	 */
+	getAttemptConditionsSubEntity() {
+		if (this._entity.hasClass(Classes.quizzes.attempts.attemptConditions) && this._entity.hasClass('collection')) {
+			return this._entity && this._entity.getSubEntityByClass(Classes.quizzes.attempts.attemptConditions);
+		}
+
+		return null;
+	}
+
+	/**
+	 * @returns {object} a singular quiz attempt condition sub-entity
+	 */
+	getAttemptConditionSubEntity(attemptConditionNumber) {
+		entity = this.getAttemptConditionsSubEntity();
+		attemptConditionEntities = entity.getSubEntitiesByClass(Classes.quizzes.attempts.attemptCondition);
+		if (!attemptConditionEntities) return;
+		attemptConditionEntities.find((entity) => {
+			if (!entity.properties || !entity.properties.attempt) return;
+			return entity.properties.attempt === attemptConditionNumber;
+		});
+
+		return null;

```

Comment:
Don't get it? Will this ever return anything other than `null`?

Reply:
Right!

Reply:
yeah, need to set the output of the `attemptConditionEntities.find(` function to something